libs/wingcompiler/src/compile.ts

```diff
@@ -168,12 +168,15 @@ export async function compile(entrypoint: string, options: CompileOptions): Prom
   const artifact = await fs.readFile(artifactPath, "utf-8");
   log?.("artifact: %s", artifact);
 
+  // TODO: is there a way to whitelist the modules we want to include?
+  const wingSDKNodeModules = join(__dirname, "..", "..", "wingsdk", "node_modules");
```

Comment:
How does this behave in a production install? This depends on the repository structure...

Also, maybe you just want to include a specific dependency here and not all of them?

You can use `require.resolve(dep)` to get its path maybe...

Reply:
I believe as long as the dependency is a bundled dependency then in the production install the resolution will still work. 
M

Reply:
It's a bad practice to rely on the fact that our dependencies are bundled. Let's find a more robust solution.

I think something like passing it the specific dependency would be better. It also means we are not exposing all of the deps internally.

Reply:
You should be able to use aliases to map known packages to their path with an [alias](https://esbuild.github.io/api/#alias):

```
alias: {
  "jsonschema": require.resolve("jsonschema"),
},
```

The most robust solution would be a small esbuild plugin, but that could wait for another time if we have issues

Reply:
I wonder if the more straight forward path is to add the "jsonschema" dependency to both wingsdk and wingcompiler and use `alias` for the preflight code and the bundle code.

Reply:
Okay so I managed to figure out how to keep this all in the sdk and not need to do any of this dependency resolution stuff here. 

The std.struct class now has a `_getValidator` method that returns a validator from the `jsonschema` library so no requires to `jsonschema` are needed in the generated code.